tools/pipelines/templates/build-npm-package.yml

```diff
@@ -1,598 +1,540 @@
-            - task: Bash@3
-              displayName: Patch Coverage Results
-              inputs:
-                targetType: 'inline'
-                workingDirectory: ${{ parameters.buildDirectory }}/nyc/report
-                script: |
-                  sed -e 's/\(filename=\".*[\\/]external .*\)"\(.*\)""/\1\&quot;\2\&quot;"/' cobertura-coverage.xml > cobertura-coverage-patched.xml
-              condition: and(succeededOrFailed(), eq(variables['ReportDirExists'], 'true'))
-            - task: PublishCodeCoverageResults@1
-              displayName: Publish Code Coverage
-              inputs:
-                codeCoverageTool: Cobertura
-                summaryFileLocation: ${{ parameters.buildDirectory }}/nyc/report/cobertura-coverage-patched.xml
-                reportDirectory: ${{ parameters.buildDirectory }}/nyc/report
-                failIfCoverageEmpty: true
-              condition: and(succeededOrFailed(), eq(variables['ReportDirExists'], 'true'))
-
-          # Publish tinylicious log for troubleshooting
-          - ${{ if contains(convertToJson(parameters.taskTest), 'tinylicious') }}:
-            - task: PublishPipelineArtifact@1
-              displayName: Publish Artifact - Tinylicious Log
-              inputs:
-                targetPath: '${{ parameters.buildDirectory }}/packages/test/test-end-to-end-tests/tinylicious.log'
-                artifactName: 'tinyliciousLog_attempt-$(System.JobAttempt)'
-                publishLocation: 'pipeline'
-              condition: and(succeededOrFailed(), eq(variables['startTest'], 'true'))
-              continueOnError: true # Keep running subsequent tasks even if this one fails (e.g. the tinylicious log wasn't there)
-
-          # Process test result, include publishing and logging
-          - template: include-process-test-results.yml
+              buildDirectory: '${{ parameters.buildDirectory }}'
+              checks: '${{ parameters.checks }}'
+              dependencyInstallCommand: pnpm install --frozen-lockfile
+      - stage: build
+        displayName: Build Stage
+        dependsOn: []
+        jobs:
+          - job: build
+            displayName: Build
+            variables:
+              testCoverage: '${{ ne(variables[''Build.Reason''], ''PullRequest'') }}'
+              releaseBuildVar: '$[variables.releaseBuild]'
+              '${{ if eq(variables[''Build.Reason''], ''PullRequest'') }}':
+                targetBranchName: $(System.PullRequest.TargetBranch)
+            steps:
+              - checkout: self
+                clean: true
+                lfs: '${{ parameters.checkoutSubmodules }}'
+                submodules: '${{ parameters.checkoutSubmodules }}'
+              - task: Bash@3
+                displayName: Parameters
+                inputs:
+                  targetType: inline
+                  workingDirectory: '${{ parameters.buildDirectory }}'
+                  script: >
```

Comment:
Based on the explanation [here](https://www.benday.com/2021/08/18/azure-devops-tip-handling-strings-in-yaml-pipelines/) around the difference between using `|` and `>`, I'm not sure this works the same; if the whole script is compressed into a single line, then the `echo` won't print a value per line but a huge unreadable line of key values.

I'd say let's keep using `|`.